crates/client/mempool/src/inner.rs

```diff
@@ -147,13 +149,13 @@ impl NonceChain {
         let mempool_tx_hash = mempool_tx.tx_hash();
 
         let replaced = if force {
-            if self.transactions.insert(OrderMempoolTransactionByNonce(mempool_tx), ()).is_some() {
+            if self.transactions.insert(OrderMempoolTransactionByNonce(mempool_tx.clone()), ()).is_some() {
```

Comment:
`clone` can be avoided by refactoring

```rust
#[cfg(debug_assertions)]
{
    self.front_tx_hash = mempool_tx.tx_hash();
}
```

```suggestion
            if self.transactions.insert(OrderMempoolTransactionByNonce(mempool_tx), ()).is_some() {
```